compiler/qsc_circuit/src/circuit_to_qsharp.rs

```diff
@@ -0,0 +1,379 @@
+        String::new()
+    } else {
+        "qs : Qubit[]".to_string()
+    };
+
+    // The return type is determined by the number of qubits "children".
+    // However, the actual return statement is determined by the variables storing measurements.
+    // If there is an inconsistency between these, which would happen if there was a mismatch between
+    // the number of qubit children specified on the circuit and the number of qubit children specified
+    // on the measurements, incorrect Q# could be generated.
+    let return_type = match circuit.qubits.iter().fold(0, |sum, q| sum + q.num_results) {
+        0 => "Unit",
+        1 => "Result",
+        _ => "Result[]",
+    };
+
+    // Check if all operations are Unitary
+    let is_ctl_adj = !circuit.component_grid.iter().any(|col| {
+        col.components
+            .iter()
+            .any(|op| !matches!(op, Operation::Unitary(_)))
+    });
+
+    let characteristics = if is_ctl_adj { "is Ctl + Adj " } else { "" };
+    let summary = if qubits.is_empty() {
+        String::new()
+    } else {
+        format!("/// Expects a qubit register of size {}.\n", qubits.len())
+    };
+
+    let mut qsharp_str = format!(
+        "{summary}operation {circuit_name}({parameter}) : {return_type} {characteristics}{{\n"
+    );
+    indentation_level += 1;
+
+    let mut measure_results = vec![];
+    let indent = "    ".repeat(indentation_level);
+
+    // Add an assert for the number of qubits
+    qsharp_str.push_str(&generate_qubit_validation(
+        circuit_name,
+        &qubits,
+        indentation_level,
+    ));
+
+    let mut body_str = String::new();
+    let mut should_add_pi = false;
+
+    // ToDo: Add support for children operations
+    for col in &circuit.component_grid {
+        for op in &col.components {
+            match &op {
+                Operation::Measurement(measurement) => {
+                    body_str.push_str(
+                        handle_measurement(measurement, &qubits, &indent, &mut measure_results)
+                            .as_str(),
+                    );
+                }
+                Operation::Unitary(unitary) => {
+                    body_str.push_str(handle_unitary(unitary, &qubits, &indent).as_str());
+                }
+                Operation::Ket(ket) => {
+                    body_str.push_str(handle_ket(ket, &qubits, &indent).as_str());
+                }
+            }
+
+            // If operation is a sqrt gate, add a π constant
+            if !should_add_pi {
+                should_add_pi = op.gate() == "SX";
+            }
+
+            // Look for a `π` character in the args
+            let args = op.args();
+            if !should_add_pi && !args.is_empty() {
+                should_add_pi = args.iter().any(|arg| arg.contains("π"));
+            }
+        }
+    }
+
+    if should_add_pi {
+        // Add the π constant
+        qsharp_str.push_str(&format!("{indent}let π = Std.Math.PI();\n"));
```

Comment:
latest clippy warns about this pattern, so use this instead:

```suggestion
        writeln!(qsharp_str, "{indent}let π = Std.Math.PI();");
```

Reply:
fixed